public/app/features/explore/QueryRow.tsx

```diff
@@ -53,7 +53,14 @@ interface QueryRowProps extends PropsFromParent {
   mode: ExploreMode;
 }
 
-export class QueryRow extends PureComponent<QueryRowProps> {
+export class QueryRow extends PureComponent<QueryRowProps, { textEditModeEnabled: boolean }> {
+  constructor(props: QueryRowProps) {
+    super(props);
+    this.state = {
```

Comment:
you do not need constructor to set initial state you can also just type state: State = { textEditModeEnabled: false } on the component level. (the type is important using this syntax)

Reply:
ah, good point, thanks!